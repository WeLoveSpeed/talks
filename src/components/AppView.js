import React, { Component } from 'react';

import TalkList from './TalkList';
import Talk from './Talk';
import Profile from './Profile';
import injectTapEventPlugin from 'react-tap-event-plugin';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';
import getMuiTheme from 'material-ui/styles/getMuiTheme'
import Drawer from 'material-ui/Drawer';
import CircularProgress from 'material-ui/CircularProgress';
import { red500 } from 'material-ui/styles/colors';
import RaisedButton from 'material-ui/RaisedButton';

// Needed for onTouchTap
// http://stackoverflow.com/a/34015469/988941
injectTapEventPlugin();

class AppView extends Component {
  static childContextTypes = {
    muiTheme: React.PropTypes.object
  }

  getChildContext() {
    return {
      muiTheme: getMuiTheme()
    }
  }

  constructor(props) {
    super(props);
    this.state = { drawerIsOpen: true }
  }

  componentDidMount() {
    window.addEventListener('google-loaded', this.props.requestAuth(true));
  }

  getErrors() {
    const {errorMessage} = this.props;
    if (errorMessage !== null) {
      return (
        <p style={{color: red500}}>{errorMessage}</p>
      )
    }
    return null;
  }

  getSelectedTalk(selectedTalk) {
    if (selectedTalk === null) {
      return (
        <main>
          <p>Veuillez s√©lectionner un talk</p>
        </main>
      )
    }

    const talk = this.props.talks[selectedTalk];

    return (
      <Talk talk={talk} />
    )
  }

  getProfile() {
    return <Profile profile={this.props.profile} signout={() => this.props.signout()} />
  }

  getContent() {
    const {
      loader,
      count, 
      selectedTalk,
      permission,
      filteredTalks
    } = this.props;
    
    if (loader) {
      return <div className="loading"><CircularProgress color={red500} size={80} thickness={5} /></div>;
    }

    if (permission === null) {
      return (
        <main>
          {this.getErrors()}
          <RaisedButton onClick={() => this.props.requestAuth()} label="Se connecter sur Google Drive" backgroundColor={red500} labelColor='white' />
        </main>
      )
    }
    return (
      <main>
        <Drawer 
          open={this.state.drawerIsOpen} 
          className='drawer'
          width={360} 
          openSecondary={true} 
          style={{position: 'relative'}}>
          <TalkList
            selectedTalk={selectedTalk}
            count={count}
            talks={filteredTalks}
            selectTalk={id => this.props.selectTalk(id)}
            setFilter={format => this.props.filterTalks(format)}
            sortBy={this.props.sortBy}
            sortTalks={value => this.props.sortTalks(value)}
            />
        </Drawer>
        {this.getSelectedTalk(selectedTalk)}
      </main>
    );
  }

  toggleDrawer = () => this.setState({drawerIsOpen: !this.state.drawerIsOpen});

  render() {
    return (
      <MuiThemeProvider>
        <div className="container">
          <AppBar
            onLeftIconButtonTouchTap={() => this.toggleDrawer()}
            style={{ backgroundColor: red500 }}
            title={<h1>Propositions de sujets Sud Web</h1>}
            iconElementRight={this.getProfile()}
            iconStyleRight={{margin: 0}}
            />
          {this.getContent()}
          <Profile />
        </div>
      </MuiThemeProvider>
    );
  }
}

export default AppView;