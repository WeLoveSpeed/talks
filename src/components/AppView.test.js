import React, { Component } from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import AppView from "./AppView";

const mockTalks = [
  {
    title: "test"
  }
];

it("renders correctly", () => {
  const component = shallow(
    <AppView
      selectedTalk={0}
      count={{ all: 3, PK: 2, PK: 1 }}
      talks={mockTalks}
    />
  );
  expect(component).toMatchSnapshot();
});
