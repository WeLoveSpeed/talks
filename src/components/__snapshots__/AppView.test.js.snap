exports[`test renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <MuiThemeProvider>
    <div
        className="container">
        <AppBar
            iconElementRight={
                <Profile
                    profile={undefined}
                    signout={[Function]} />
            }
            iconStyleRight={
                Object {
                    "margin": 0,
                  }
            }
            onLeftIconButtonTouchTap={[Function]}
            showMenuIconButton={true}
            style={
                Object {
                    "backgroundColor": "#f44336",
                  }
            }
            title={
                <h1>
                    Propositions de sujets We Love Speed (
                    <a
                        href="https://docs.google.com/spreadsheets/d/1AjbdFcj2aebJ_lExkBryB-SC3O7ht8RHq7BL_1aw_LQ/edit#gid=237957982"
                        target="_blank">
                        sheet
                    </a>
                    )
                </h1>
            }
            zDepth={1} />
        <main>
            <Drawer
                className="drawer"
                disableSwipeToOpen={false}
                docked={true}
                open={true}
                openSecondary={false}
                style={
                    Object {
                        "position": "relative",
                      }
                }
                swipeAreaWidth={30}
                width={360}
                zDepth={2}>
                <TalkList
                    count={
                        Object {
                            "PK": 1,
                            "all": 3,
                          }
                    }
                    selectTalk={[Function]}
                    selectedTalk={0}
                    setFilter={[Function]}
                    sortBy={undefined}
                    sortTalks={[Function]}
                    talks={undefined} />
            </Drawer>
            <Talk
                talk={
                    Object {
                        "title": "test",
                      }
                } />
        </main>
        <Profile />
    </div>
</MuiThemeProvider>,
  "nodes": Array [
    <MuiThemeProvider>
      <div
            className="container">
            <AppBar
                  iconElementRight={
                        <Profile
                              profile={undefined}
                              signout={[Function]} />
                  }
                  iconStyleRight={
                        Object {
                              "margin": 0,
                            }
                  }
                  onLeftIconButtonTouchTap={[Function]}
                  showMenuIconButton={true}
                  style={
                        Object {
                              "backgroundColor": "#f44336",
                            }
                  }
                  title={
                        <h1>
                              Propositions de sujets We Love Speed (
                              <a
                                    href="https://docs.google.com/spreadsheets/d/1AjbdFcj2aebJ_lExkBryB-SC3O7ht8RHq7BL_1aw_LQ/edit#gid=237957982"
                                    target="_blank">
                                    sheet
                              </a>
                              )
                        </h1>
                  }
                  zDepth={1} />
            <main>
                  <Drawer
                        className="drawer"
                        disableSwipeToOpen={false}
                        docked={true}
                        open={true}
                        openSecondary={false}
                        style={
                              Object {
                                    "position": "relative",
                                  }
                        }
                        swipeAreaWidth={30}
                        width={360}
                        zDepth={2}>
                        <TalkList
                              count={
                                    Object {
                                          "PK": 1,
                                          "all": 3,
                                        }
                              }
                              selectTalk={[Function]}
                              selectedTalk={0}
                              setFilter={[Function]}
                              sortBy={undefined}
                              sortTalks={[Function]}
                              talks={undefined} />
                  </Drawer>
                  <Talk
                        talk={
                              Object {
                                    "title": "test",
                                  }
                        } />
            </main>
            <Profile />
      </div>
</MuiThemeProvider>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AppView
        count={
                Object {
                        "PK": 1,
                        "all": 3,
                      }
        }
        selectedTalk={0}
        talks={
                Array [
                        Object {
                          "title": "test",
                        },
                      ]
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AppView {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "count": Object {
            "PK": 1,
            "all": 3,
          },
          "selectedTalk": 0,
          "talks": Array [
            Object {
              "title": "test",
            },
          ],
        },
        "refs": Object {},
        "state": Object {
          "drawerIsOpen": true,
        },
        "toggleDrawer": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <MuiThemeProvider>
          <div
                    className="container">
                    <AppBar
                              iconElementRight={
                                        <Profile
                                                  profile={undefined}
                                                  signout={[Function]} />
                              }
                              iconStyleRight={
                                        Object {
                                                  "margin": 0,
                                                }
                              }
                              onLeftIconButtonTouchTap={[Function]}
                              showMenuIconButton={true}
                              style={
                                        Object {
                                                  "backgroundColor": "#f44336",
                                                }
                              }
                              title={
                                        <h1>
                                                  Propositions de sujets We Love Speed (
                                                  <a
                                                            href="https://docs.google.com/spreadsheets/d/1AjbdFcj2aebJ_lExkBryB-SC3O7ht8RHq7BL_1aw_LQ/edit#gid=237957982"
                                                            target="_blank">
                                                            sheet
                                                  </a>
                                                  )
                                        </h1>
                              }
                              zDepth={1} />
                    <main>
                              <Drawer
                                        className="drawer"
                                        disableSwipeToOpen={false}
                                        docked={true}
                                        open={true}
                                        openSecondary={false}
                                        style={
                                                  Object {
                                                            "position": "relative",
                                                          }
                                        }
                                        swipeAreaWidth={30}
                                        width={360}
                                        zDepth={2}>
                                        <TalkList
                                                  count={
                                                            Object {
                                                                      "PK": 1,
                                                                      "all": 3,
                                                                    }
                                                  }
                                                  selectTalk={[Function]}
                                                  selectedTalk={0}
                                                  setFilter={[Function]}
                                                  sortBy={undefined}
                                                  sortTalks={[Function]}
                                                  talks={undefined} />
                              </Drawer>
                              <Talk
                                        talk={
                                                  Object {
                                                            "title": "test",
                                                          }
                                        } />
                    </main>
                    <Profile />
          </div>
</MuiThemeProvider>,
        "_debugID": 2,
        "_renderedOutput": <MuiThemeProvider>
          <div
                    className="container">
                    <AppBar
                              iconElementRight={
                                        <Profile
                                                  profile={undefined}
                                                  signout={[Function]} />
                              }
                              iconStyleRight={
                                        Object {
                                                  "margin": 0,
                                                }
                              }
                              onLeftIconButtonTouchTap={[Function]}
                              showMenuIconButton={true}
                              style={
                                        Object {
                                                  "backgroundColor": "#f44336",
                                                }
                              }
                              title={
                                        <h1>
                                                  Propositions de sujets We Love Speed (
                                                  <a
                                                            href="https://docs.google.com/spreadsheets/d/1AjbdFcj2aebJ_lExkBryB-SC3O7ht8RHq7BL_1aw_LQ/edit#gid=237957982"
                                                            target="_blank">
                                                            sheet
                                                  </a>
                                                  )
                                        </h1>
                              }
                              zDepth={1} />
                    <main>
                              <Drawer
                                        className="drawer"
                                        disableSwipeToOpen={false}
                                        docked={true}
                                        open={true}
                                        openSecondary={false}
                                        style={
                                                  Object {
                                                            "position": "relative",
                                                          }
                                        }
                                        swipeAreaWidth={30}
                                        width={360}
                                        zDepth={2}>
                                        <TalkList
                                                  count={
                                                            Object {
                                                                      "PK": 1,
                                                                      "all": 3,
                                                                    }
                                                  }
                                                  selectTalk={[Function]}
                                                  selectedTalk={0}
                                                  setFilter={[Function]}
                                                  sortBy={undefined}
                                                  sortTalks={[Function]}
                                                  talks={undefined} />
                              </Drawer>
                              <Talk
                                        talk={
                                                  Object {
                                                            "title": "test",
                                                          }
                                        } />
                    </main>
                    <Profile />
          </div>
</MuiThemeProvider>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AppView
    count={
        Object {
            "PK": 1,
            "all": 3,
          }
    }
    selectedTalk={0}
    talks={
        Array [
            Object {
              "title": "test",
            },
          ]
    } />,
}
`;
